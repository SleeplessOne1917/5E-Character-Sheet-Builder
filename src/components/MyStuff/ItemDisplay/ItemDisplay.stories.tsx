import { ComponentMeta, ComponentStory } from '@storybook/react';

import ItemDisplay from './ItemDisplay';

export default {
	title: 'Components/MyStuff/Item',
	component: ItemDisplay,
	args: {
		item: { id: 'foo', name: 'Foo' }
	}
} as ComponentMeta<typeof ItemDisplay>;

const Template: ComponentStory<typeof ItemDisplay> = args => (
	<ItemDisplay {...args} />
);

export const Default = Template.bind({});
