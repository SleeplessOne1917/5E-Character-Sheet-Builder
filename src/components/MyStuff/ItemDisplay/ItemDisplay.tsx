import Button from '../../Button/Button';
import { Item } from '../../../types/db/item';
import classes from './ItemDisplay.module.css';
import { useCallback } from 'react';

type ItemDisplayProps = {
	item: Item;
	onMoreInfoClick: (item: Item) => void;
};

const ItemDisplay = ({ item, onMoreInfoClick }: ItemDisplayProps) => {
	const handleMoreInfoClick = useCallback(
		() => onMoreInfoClick(item),
		[onMoreInfoClick, item]
	);

	return (
		<div className={classes.item} data-testid="item">
			<div className={classes.info}>{item.name}</div>
			<Button size="small" positive onClick={handleMoreInfoClick}>
				More Info
			</Button>
		</div>
	);
};

export default ItemDisplay;
