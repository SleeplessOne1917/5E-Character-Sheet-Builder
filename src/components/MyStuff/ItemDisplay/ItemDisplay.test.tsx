import * as stories from './ItemDisplay.stories';

import { render, screen } from '@testing-library/react';

import { composeStories } from '@storybook/react';

const { Default } = composeStories(stories);

it('renders correctly', () => {
	render(<Default />);

	expect(screen.getByTestId('item')).toMatchSnapshot();
});
