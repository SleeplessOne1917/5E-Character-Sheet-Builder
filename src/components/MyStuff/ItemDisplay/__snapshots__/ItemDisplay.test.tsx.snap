// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  class="item"
  data-testid="item"
>
  <div
    class="info"
  >
    Foo
  </div>
  <button
    class="button positive"
    style="font-size: 0.7rem; padding: 0.3rem; margin: 0.25rem;"
    type="button"
  >
    More Info
  </button>
</div>
`;
